### PR TITLE
Fix/refactoring search

### PR DESCRIPTION
**Pull request by brujeo**
An issue with the search filter has been discovered, so this PR aims to:
- refactor search filter 
- switch response from XML to JSON to simplify communication and parsing and make it less error-prone
- switch EA API version from 1.0.0. to 2.0.0

### CODE REVIEW

File: lib/defra_ruby/area/area.rb [hunk 1]
```diff
@@ -7,24 +7,30 @@ module Area
...
+        end
       end
 
-      def parse_xml
-        @area_id = @xml.xpath("*[local-name()='identifier']").text.to_i
-        @code = @xml.xpath("*[local-name()='code']").text
-        @long_name = @xml.xpath("*[local-name()='long_name']").text
-        @short_name = @xml.xpath("*[local-name()='short_name']").text
+      def parse_area_record
+        @area_id = @area_record["properties"]["identifier"].to_i
```
Comment (jjromeo) [inline lib/defra_ruby/area/area.rb]: Could use #dig

File: lib/defra_ruby/area/version.rb [hunk 2]
```diff
@@ -2,6 +2,6 @@
 
 module DefraRuby
   module Area
-    VERSION = "2.1.0"
+    VERSION = "2.2.0"
```
Comment (jjromeo) [inline lib/defra_ruby/area/version.rb:5]: Just to confirm, this is minor due to the changes all being behind the scenes and not actually affecting the end user app?

**Review by jjromeo — DISMISSED** [review]

File: lib/defra_ruby/area/version.rb (hunk 2 above)
Comment (brujeo) [reply lib/defra_ruby/area/version.rb:5]: This is quite major. EA API version has been switched from 1.0.0 to 2.0.0, search filter has been fully rewritten, response format switched from XML to JSON. I've tried to make it backwards-compatible, although that's not guaranteed

File: lib/defra_ruby/area/area.rb (hunk 1 above)
Comment (brujeo) [reply lib/defra_ruby/area/area.rb]: refactored

**Review by jjromeo — APPROVED** [review]

**Pull request merged**